### ISSUE TITLE
Simple question about the att- decoder loss

### ISSUE THREAD

**Issue by TParcollet** [espnet/espnet#3291]
Hi everyone,

I am wondering why you are multiplying the loss by (from what I understand) the mean of the length of the targets minus 1:
`self.loss *= np.mean([len(x) for x in ys_in]) - 1` From my understanding of the code, the loss is already averaged before, hence scaled-down w.r.t the length of sequence, right ?

Here: https://github.com/espnet/espnet/blob/1f8f745d6abb760984e22a67396c708dea738a7e/espnet/nets/pytorch_backend/rnn/decoders.py#L281

Thank you!

**Comment by sw005320** [comment]
> scaled-down w.r.t the length of sequence, right?

Yes, and this makes it log p(W_{1:N}|O_{1:T}) instead of (log p(W_{1:N}|O_{1:T}))/N (unnormalized by the length) so that we can combine it with the CTC loss with the same domain (which is also log p(W|O))

**Comment by TParcollet** [comment]
Hi Shinji and thanks for the answer. I Guess I am not familiar enough with the format of the tensors in this part of the code. Isn't:
`self.loss = F.cross_entropy(y_all,ys_out_pad.view(-1),ignore_index=self.ignore_id,reduction=reduction_str)`
Already averaging over the timesteps + batch? 

Thanks.

**Comment by sw005320** [comment]
Yes, this part averages over the "timesteps + batch".
Then, we multiplied the average timestep length to make the loss averaged only with "batch". 
We don't want to average over "timesteps" to combine it with CTC.

**Comment by TParcollet** [comment]
I see, thanks!
